src/arrayMethodJoin.js

```diff
@@ -4,9 +4,37 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let separatorString;
+    let finalString = '';
+
+    switch (separator) {
+      case null: separatorString = 'null';
+        break;
+      case undefined: separatorString = 'undefined';
+        break;
+      case '': separatorString = '';
```

Comment:
You can just check if it's not null and undefined, and add a separator to the string only in that case.
There is really no need to use switch case